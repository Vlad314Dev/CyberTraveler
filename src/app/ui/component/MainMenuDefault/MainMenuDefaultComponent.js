import './MainMenuDefaultStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import {
    AUTHORIZE_OPTION,
    optionLabels,
    PLAY_GAME_OPTION,
    SETTINGS_OPTION,
    VIEW_LEADERBOARD_OPTION
} from 'UIStore/MainMenu/MainMenuConfig';

class MainMenuDefaultComponent extends PureComponent
{
    static propTypes = {
        setActiveOption: PropTypes.func.isRequired,
        activeOption: PropTypes.string.isRequired
    };

    setActiveOption = (optionName) => () => {
        const { setActiveOption } = this.props;
        setActiveOption(optionName)
    }

    getActiveOptionLabel()
    {
        const { activeOption } = this.props;
        const option = optionLabels.find((data) => data.key === activeOption);

        if (option) {
            return option.label;
        }

        return null;
    }

    render()
    {   
        return (
            <div block="MainMenuOption" elem="List">
                {optionLabels.map((option, index) => {
                    if (option.key === AUTHORIZE_OPTION 
                        || option.key === PLAY_GAME_OPTION
                        || option.key === SETTINGS_OPTION 
                        || option.key === VIEW_LEADERBOARD_OPTION
                    ) {
                        return <div 
                            block="MainMenuOption" 
                            elem="Wrapper" 
                            key={ index }
                        >
                            <span 
                                block="MainMenuOption" 
                                elem="Value"
                                key={ index }
                                onClick={ this.setActiveOption(option.key) }
                            >{ option.label }</span>
                        </div>
                    }
                    
                    return null;
                })}
            </div>
        );
    }
}

export default MainMenuDefaultComponent;
