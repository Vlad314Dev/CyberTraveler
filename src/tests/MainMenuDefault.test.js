/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import MainMenuDefaultComponent from '../app/ui/component/MainMenuDefault/MainMenuDefaultComponent';

describe('MainMenuDefault', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <MainMenuDefaultComponent 
                activeOption={ '' }
                setActiveOption={ jest.fn() }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <MainMenuDefaultComponent 
                activeOption={ '' }
                setActiveOption={ jest.fn() }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
