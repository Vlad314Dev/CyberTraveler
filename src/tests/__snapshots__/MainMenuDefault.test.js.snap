// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuDefault Should render layout 1`] = `
Array [
  <div
    className="MainMenuOption__List"
  >
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Play
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Leaderboard
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Authorize
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Settings
      </span>
    </div>
  </div>,
]
`;
